### ISSUE TITLE
Compiling ics-openvpn really feels like MISSION IMPOSSIBLE regardless of what OS I try to do it on Windows or Linux

### ISSUE THREAD

**Issue by DreamSeekerKaiser** [schwabe/ics-openvpn#1047]
Hi there,

I've literally been trying to sort this out for more than 4 weeks now without any kind of success.

I am trying to compile the ics-openvpn (https://github.com/schwabe/ics-openvpn) for but I am having a hard time. I tried at first on my windows machine, but then said why'd I go through all of this hassle, so just fired up my Ubuntu VM but guess what? More issues, lol.

Basically I clone the repository, then run:

git submodule init  
git submodule update 
After that I open the project using Android Studio, but the gradle sync fails (asking me for a gradle wrapper saying it's not found?) I am just not so familiar with the whole building process of apps, I just know how to code, so my experience with Gradle and other building stuff is kinda meh.

I'd like it if someone could put me in the right direction here. I've read a lot of articles showing how to build it, watched some videos on Youtube, but everything is outdated.

Thanks in advance.

**Comment by pep1439** [comment]
Hi, i also trying to setup it on windows machine, but facing problem doing so. 
Cloned after setting up the ssh key from : git@github.com:schwabe/ics-openvpn.git 
git submodule init
git submodule update
git checkout v0.7.5 //after this i can only see the \ics-openvpn\main\misc\build-native.bat
moved to \ics-openvpn\main\misc\
then calling build-native and getting below:
'Cannot find .git directory in openvpn, aborting'

also i tried to call this bat file from \ics-openvpn\main\src\main\cpp as it contains the openvpn dir and getting the below huge output:

C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp>C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\misc\build-native.bat
Android NDK: android-14 is unsupported. Using minimum supported version android-16.
C:/Users/amit.agrahari/AndroidStudioProjects/Schwabe/ics-openvpn/main/jni/Android.mk:43: openvpn/Android.mk: No such file or directory
C:/Users/amit.agrahari/AppData/Local/Android/Sdk/ndk-bundle/build//../build/core/build-binary.mk:666: Android NDK: Module nopie_openvpn depends on undefined modules: openvpn
C:/Users/amit.agrahari/AppData/Local/Android/Sdk/ndk-bundle/build//../build/core/build-binary.mk:679: *** Android NDK: Aborting (set APP_ALLOW_MISSING_DEPS=true to allow missing dependencies)    .  Stop.
The system cannot find the path specified.
A subdirectory or file ..\ovpnlibs already exists.
A subdirectory or file ..\ovpnlibs\assets already exists.
A subdirectory or file ..\ovpnlibs\jniLibs already exists.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\asio\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\asio\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\asio\ already exists.
asio\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\jbcrypto\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\jbcrypto\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\jbcrypto\ already exists.
jbcrypto\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\lz4\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\lz4\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\lz4\ already exists.
lz4\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\lzo\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\lzo\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\lzo\ already exists.
lzo\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\mbedtls\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\mbedtls\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\mbedtls\ already exists.
mbedtls\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\minivpn\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\minivpn\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\minivpn\ already exists.
minivpn\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openssl\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openssl\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\openssl\ already exists.
openssl\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openvpn\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openvpn\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\openvpn\ already exists.
openvpn\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openvpn-config\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openvpn-config\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\openvpn-config\ already exists.
openvpn-config\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\Schwabe\ics-openvpn\main\src\main\cpp\openvpn3\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\openvpn3\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\openvpn3\ already exists.
openvpn3\*.so
The system cannot find the file specified.
        0 file(s) copied.
The system cannot find the file specified.
The system cannot find the file specified.
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\opvpnutil\libcrypto.so
Could Not Find C:\Users\a\AndroidStudioProjects\Schwabe\ics-openvpn\main\src\main\cpp\opvpnutil\libssl.so
A subdirectory or file ..\ovpnlibs\jniLibs\opvpnutil\ already exists.
opvpnutil\*.so
The system cannot find the file specified.
        0 file(s) copied.


Any idea how to do this in Windows?
Regards.

**Comment by schwabe** [comment]
Windows should work. But the repo IS buildable, see the travis build: https://travis-ci.org/schwabe/ics-openvpn

**Comment by pep1439** [comment]
Hi Schwabe, thanks for the reply, but as you can see in my comment, i have done the right thing but no success :( , could you please elaborate more.
I downloaded the .apk file and seems working fine on my device and just wanted to run and check the src code on my Windows system. I will use this in commercial purpose too. Please help.

**Comment by pep1439** [comment]
Hi Schwabe,
Its working fine now, had missed couple of things.
Thanks.

**Comment by schwabe** [comment]
```build-native.bat``` is not the repo anymore. The instrucstion in the README are all that you really need.

**Comment by pep1439** [comment]
Yes. Thanks.

**Comment by pep1439** [comment]
Hi Schwabe, 
Want to use your code in my app(commercial app) to build a VPN client to connect a private network, i dont know how to contact you, can you say hi to me at pep1439@gmail.com. Want to discuss on licensing. 

Thanks,
Amit

**Comment by schwabe** [comment]
arne-openvpn@rfc2549.org

**Comment by okcomputer** [comment]
> Windows should work. But the repo IS buildable, see the travis build: https://travis-ci.org/schwabe/ics-openvpn

All I want to say is thank you. I was trying to build this repo for 12 hours today. I started to look through closed issues to see if anyone has faced the same issues or felt like I was. I saw this issue which and thought "man I feel the same way".

Then after reading this comment it dawned to me to look at the travis-ci build steps and in less than one hour I can confirm it's building great!!! So Thank you!!
